app/ui/src/components/Register.vue

```diff
@@ -84,35 +88,54 @@ export default {
       username: '',
       password: '',
       passwordConfirm: '',
-      email: ''
+      email: '',
+      agreeWithTermsAndConfitions: false
     }
   },
   methods: {
     submitRegistration () {
       this.errors = []
       this.submittingRegistration = true
-      var data = {
-        username: this.username,
-        password: this.password,
-        passwordConfirm: this.passwordConfirm,
-        email: this.email
-      }
-      axios.post('http://127.0.0.1:5000/api/v1/registration', data)
-        .then(response => {
-          // Json response
-          // var items = []
-          // response.data.forEach(element => {
-          //   items.push(element['value'])
-          // })
-          // this.msg = 'Welcome to your Vue.js App. ' + items.join(', ')
-          this.errors.push('Success!')
-        })
-        .catch(e => {
-          e.response.data['messages'].forEach(message => {
-            this.errors.push(message)
+      if (this.isFormValid()) {
+        var data = {
+          username: this.username,
+          password: this.password,
+          passwordConfirm: this.passwordConfirm,
+          email: this.email
+        }
+        axios.post('http://127.0.0.1:5000/api/v1/registration', data)
+          .then(response => {
+            this.errors.push('Success!')
+          })
+          .catch(e => {
+            e.response.data['messages'].forEach(message => {
+              this.errors.push(message)
+            })
           })
-        })
+      }
       this.submittingRegistration = false
+    },
+    isFormValid () {
+      if (this.username.length === 0) {
+        this.errors.push('Username is a required field.')
+      }
+      if (this.password.length === 0) {
+        this.errors.push('Password is a required field.')
+      }
+      if (this.passwordConfirm.length === 0) {
+        this.errors.push('Password Confirmation is a required field.')
+      }
+      if (this.password !== this.passwordConfirm) {
+        this.errors.push('Password and Password Confirmation must match.')
+      }
+      if (this.email.length === 0) {
+        this.errors.push('Email is a required field.')
+      }
+      if (!this.agreeWithTermsAndConfitions.checked) {
+        this.errors.push('You must agree with the Terms and Conditions to register with this site.')
```

Comment:
We should create a story/task to localize these strings. Nothing to fix for this PR as I believe we have the same error on the Login page.